### PR TITLE
Rearrange

### PR DESCRIPTION
**Pull request by montemishkin**
This does not include any database or authentication, etc.  It is just the rearrangement of the generated project's file structure.  

Any other movements you wanted to make?

### CODE REVIEW

File: generators/project/templates/NAME/src/apps/client/routes.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// local imports
+import App from './views/App'
+import NotFound from './views/NotFound'
+import Home from './views/Home'
+
+
+export default (
+    <Route path='/' component={App}>
+        <IndexRoute component={Home} />
+        <Route path='*' component={NotFound} />
```
Comment (aaivazis) [inline generators/project/templates/NAME/src/apps/client/routes.js]: Do the routes resolve in order? How do we implement multiple views if the nonFound path is '*'

Comment (montemishkin) [inline generators/project/templates/NAME/src/apps/client/routes.js]: I believe they resolve in order.  So you just need to make sure that the catch all route is always last.

Comment (montemishkin) [inline generators/project/templates/NAME/src/apps/client/routes.js]: or did i misinterpret your question?

**Comment by aaivazis** [comment]
*sigh* that's really fragile. I wish there was still a NotFoundRoute

**Comment by montemishkin** [comment]
If you find a better way I would love to see it.  This is what I ended up with after looking into it for a while though.  I agree its not ideal, but its also not too terrible.

I should note that with this set up of having a catch all on the client router, it means that [this](https://github.com/skycrest/generator-nautilus/blob/eefc5bd9acbfff182d973d3898ff3804f59f9dab/generators/project/templates/NAME/src/apps/client/index.js#L52-L63) case will (should) never happen.

**Comment by aaivazis** [comment]
What's wrong with rendering it from the server? I really hate relying on the frontend to handle 404s.

**Comment by aaivazis** [comment]
It's pretty terrible, imo. The server has no way of responding with a 404 if we were using another client like an ios app

**Comment by aaivazis** [comment]
I guess my main issue is that according to the server, the request was valid. That makes me really uncomfortable

**Comment by montemishkin** [comment]
If I understand things correctly, here is what would happen if you were to leave everything as is, except take away the catch all in the client (react-router) router.

### Situation 1: A request is made to the server for some not found route.
- everything goes as one would expect.  [this](https://github.com/skycrest/generator-nautilus/blob/eefc5bd9acbfff182d973d3898ff3804f59f9dab/generators/project/templates/NAME/src/apps/client/index.js#L52-L63) block gets executed and the response goes back with status 404, but with a nicely rendered 404 view nested into the app's UI.
- once the client receives the markup and begins to execute the client js entry, it passes control to react router, who sees that the current window.location is not a valid route.  react router throws an error :(

### Situation 2: A transition is made within the client single page app which tries to take the user to a not found location
- react router sees that the location is not matched by any route and thus throws an error :(

In both cases, the thrown error results from react router not being able to match the location to any of the routes.  This is "solved" by including a catch all route which renders the 404 page.

**Comment by montemishkin** [comment]
You have a good point about the request (undesirably) being valid to the server even though it is a not found route and should get status 404.  I'm currently not sure what the solution is but would like to hear of any alternate ideas.

**Comment by aaivazis** [comment]
I have no problem with having a NotFound route on the frontend, and if that means having a wildcard as the path, then that's a really big disappointment from react-router. But even still we should **not** be replying to  invalid requests with a 200.

**Comment by aaivazis** [comment]
Ideally, the only 404s that get passed onto the frontend are GETs. And even that has some weird behavior when you think about the backend as primarily serving some sort of API.

**Comment by montemishkin** [comment]
Here's react-router stance on it (not that i am defending them):
https://github.com/rackt/react-router/blob/master/UPGRADE_GUIDE.md#notfound-route

**Comment by montemishkin** [comment]
relevant issue:
https://github.com/rackt/react-router/issues/458

**Comment by montemishkin** [comment]
https://github.com/rackt/react-router/issues/1103

**Pull request merged**